packages/Search/test/CMakeLists.txt

```diff
@@ -11,11 +11,22 @@ TRIBITS_ADD_EXECUTABLE_AND_TEST(
   )
 TRIBITS_ADD_EXECUTABLE_AND_TEST(
   Exception
-  SOURCES tstException.cpp unit_test_main.cpp
+  SOURCES tstException.cpp
   COMM serial mpi
-  STANDARD_PASS_OUTPUT
-  FAIL_REGULAR_EXPRESSION "data race;leak;runtime error"
+  ADDED_EXE_TARGET_NAME_OUT dbc_exe
   )
+# TODO link Boost::dynamic_linking interface target to enable dynamic linking
+# (adds BOOST_ALL_DYN_LINK)
+target_compile_definitions(${dbc_exe} PRIVATE BOOST_TEST_MAIN BOOST_TEST_DYN_LINK)
```

Comment:
so each of these `target_compile_definitions` links differently and ultimately changes how `main()` is generated. are you trying to decide here which one you want to use?

Reply:
> so each of these `target_compile_definitions` links differently and ultimately changes how `main()` is generated.

exactly

> are you trying to decide here which one you want to use?

I will use all versions depending on the unit tests.  I might also link several unit tests into one executable.  Still figuring that one out.